gax-grpc/BUILD.bazel

```diff
@@ -27,6 +27,7 @@ _COMPILE_DEPS = [
     "@io_grpc_grpc_java//context:context",
     "@io_grpc_grpc_netty_shaded//jar",
     "@io_grpc_grpc_grpclb//jar",
+    "@io_grpc_grpc_rls//jar",
```

Comment:
Thanks for pointing out. I am not familiar with adding a dependency in Java, so I am just learning from existing examples. Does the PR look good now? 

Since this is a runtime dependency, no change in .java is needed. I am adding RLS dependency in gax-java instead of java-bigtable or java-spanner because `io_grpc_grpc_googleapis` is already added to gax-java, so this might be a proper place.

Reply:
`gax-java` does not use `io_grpc_grpc_rls` directly. This is for DirectPath RLS, as for now only Bigtable and Spanner has a plan to migrate to DirectPath, I think only Bigtable and Spanner will use it.

For the `grpc-bom` in [shared dependency bom](https://github.com/googleapis/java-shared-dependencies/blob/main/first-party-dependencies/pom.xml), from [here](https://mvnrepository.com/artifact/io.grpc/grpc-bom/1.46.0) it already involves grpc-googleapis, grpc-grpclb, grpc-rls, which are runtime dependencies needed for DirectPath. Does this mean grpc-rls already has been added? Why I still can find runtime dependency for [io_grpc_grpc_googleapis](https://github.com/googleapis/gax-java/blob/main/gax-grpc/build.gradle#L23)?

I am thinking adding `io_grpc_grpc_rls` to `gax-java` because `gax-java` help client libraries to manage network details (channel creation, channel management, retry, etc), and `io_grpc_grpc_rls` is a network feature, so `gax-java` should help client libraries to manage it?

Reply:
> gax-java does not use io_grpc_grpc_rls directly. This is for DirectPath RLS, as for now only Bigtable and Spanner has a plan to migrate to DirectPath, I think only Bigtable and Spanner will use it.

If that's the case, I would strongly suggest only adding these two to the individual repos. For maven projects, there are usually two steps to manage dependencies, first declare it in the [dependency management section](https://github.com/googleapis/java-shared-dependencies/blob/b0a30cdecae8dcfb3a6d36e3f194c39a0086673e/first-party-dependencies/pom.xml#L75), then reference it in dependency section in the respective repo. So in this case, `grpc-rls` is already part of the bom(meaning declared but not used yet), but you still need to reference it in `java-spanner` or `java-bigtable`, and make it `runtime` scope. It would be very similar to this [grpc-grpclb](https://github.com/googleapis/java-bigtable/blob/22acfc405e23ccfeeeedcce6992f887d48d22bd5/google-cloud-bigtable/pom.xml#L173) dependency in bigtable, which is not in gax-java but a runtime dependency in bigtable.

For `io_grpc_grpc_googleapis`, I don't know the history about it. I approved it last time because it is a replacement for xds, which I thought is a required runtime dependency in gax-java. If the usage of `io_grpc_grpc_googleapis` is the same as  `io_grpc_grpc_rls`, we should probably move it too.

Reply:
I see, so I will move both `io_grpc_grpc_googleapis` and `io_grpc_grpc_rls` to java-bigtable and java-spanner. Thanks!

Reply:
Thanks a lot!